js/playlist.js

```diff
@@ -21,36 +23,41 @@ function(ytData,ytIframe,template,$) {
         } else {
             if(JSON.parse(localStorage.getItem('new_playlist_saved'))) {
                 playlist = JSON.parse(localStorage.getItem('playlist'));
-                var rptst = localStorage.getItem('repeat');
-
-                total_duration = JSON.parse(localStorage.getItem('total_duration'));
+                if (curr_id==undefined) curr_id=playlist.length-1;
+                repeat = JSON.parse(localStorage.getItem('repeat'));
+                var rptst = repeat[curr_id];
+                durations=JSON.parse(localStorage.getItem('duration'));
+                total_duration = durations[curr_id];
                 updateDurationOnDisplay();
+                addPlaylistDropdown();
 
                 if(rptst=='one')
                     repeat_one=true;
                 else if(rptst=='all')
                     repeat_all=true;
                 else if(rptst=='shuffle')
                     shuffle=true;
-
                 console.log("loaded playlist");
             }
         }
-
-        // Display all playlist songs
-        for(var i=0; i<playlist.length; i++){
-            var vidrow=template.getVidRow(playlist[i]);
-            var id=playlist[i].id.videoId;
-            $(vidrow).addClass("item-video waves-effect");
-            $(vidrow).on("click", function(id){return function(event){
-                require('player').playSong(id);
-            }}(id));
-            $(vidrow).find("#action").on("click",function(id){return function(event){
-                removeFromPlaylist(event,id);
-            }}(id));
-            $(vidrow).find("#action i").html("clear");
-            vidrow.appendTo("#playlist");
+        if (playlist.length!=0){
```

Comment:
To avoid these checks and unnecessary crashes like the one I pointed to above, ensure that there is atleast one playlist at all times in the playlist array, even if its an empty one. i.e. `playlist = [[]]`. Never let the `playlist.length` become 0. It seems pointless to care for a case when there are "no playlists", instead that case can be considered as one empty playlist.